components/eam/src/chemistry/utils/modal_aero_calcsize.F90

```diff
@@ -1484,9 +2139,9 @@ subroutine modal_aero_calcsize_diag(state, pbuf, list_idx_in, dgnum_m)
          call endrun('modal_aero_calcsize_diag called for'// &
                      'diagnostic list but dgnum_m pointer not present')
       end if
-      allocate(dgnum_m(pcols,pver,nmodes), stat=stat)
-      if (stat > 0) then
-         call endrun('modal_aero_calcsize_diag: allocation FAILURE: dgnum_m')
+      if(.not. associated(dgnum_m)) then
```

Comment:
I am reworking this logic where an error is issued if this variable is not allocated already.

Reply:
Great. You might also check whether you can make dgnum_m and other arrays `allocatable` rather than `pointer`. That would provide code-level protection against future memory leaks.

Reply:
That's a great point. I have reworked the logic to use allocatable arrays in calcsize and wateruptake. I am no longer doing any allocation within the calsize_diag routine.